pkg/collector/corechecks/servicediscovery/module/impl_linux_test.go

```diff
@@ -356,25 +356,67 @@ func buildFakeServer(t *testing.T) string {
 	return binDir
 }
 
-func TestAPMInstrumentationProvided(t *testing.T) {
-	serverDir := buildFakeServer(t)
-	url := setupDiscoveryModule(t)
+func makeFakeNodeWorkingDirectory(t *testing.T) string {
```

Comment:
Maybe simpler to just check in a package.json into a testdata directory and point the process at it instead of creating the file at runtime?